docroot/modules/humsci/hs_layouts/hs_layouts.module

```diff
@@ -75,13 +75,25 @@ function hs_layouts_preprocess_pattern_accordion(&$variables) {
   }
 }
 
-// Helper function to check if a title has an element_type.
+/**
+ * Helper function to check if a title has an element_type.
+ */
 function _hs_layouts_field_has_title($field) {
   // if the field has an element_type that is a heading, return TRUE to
   // indicate that the title should be wrapped in a heading tag.
   return isset($field->element_type) && preg_match('/h[1-6]/', $field->element_type);
 }
 
+/**
+ * Helper function to expose node url when pattern is being used in a view.
+ */
+function _hs_layouts_view_add_node_url(&$variables) {
+  if ($variables['context']->isOfType('views_row')) {
+    $node = $variables['context']->getProperty('row')->_entity;
+    $variables['node_url'] = $node->toUrl()->toString();
+  }
+}
```

Comment:
Instead of doing all this every single time for every pattern, just use display suite's "Extra Fields" at `/admin/structure/ds/fields` then remove all this unnecessary stuff in favor of UI config changes.

Reply:
@pookmish Thanks for the feedback! I don't think a DS field would work in this case, because what we need is to access data from the node object from the pattern template in a view. However I agree that adding this to the preprocess is not the best approach so we updated the template files to take care of everything there. PR is ready for review again, please let us know if there's anything else to fix.

Reply:
ah, ds fields would help if it were used in display modes. FWIW, my team use display modes instead of view fields whenever possible. I'm ok with the new solution at this time..